### DISCUSSION TITLE
OAuth2 "password" grant type

### DISCUSSION THREAD

**Discussion by vadrem** [graphql-java-kickstart/graphql-spring-webclient#28] [Q&A] [answered]
Hello,
I would like to know does this project support the ability to use "password" grant type?
By default is "client_credentials" is  used oauth2.authorization-grant-type=client_credentials.
If yes, then where (how) can I to set username and password for token request?

**Comment by oliemansm** [comment]
Password grant type is not something you'd use for a backend integration. That's what you would use when you have an actual user trying to login with username and password. So you also wouldn't have that username and password as properties, but instead would have some kind of UI where the user would enter those.

**Comment by vadrem** [comment]
i assumed my one BE service 1 is a client for another BE service 2 (resource server on graphql-java-kickstart/graphql-spring-boot). Accordingly, service 1 keeps in itself username and password, and could receive token without UI.
For example: https://docs.spring.io/spring-security/site/docs/current/reference/html5/#using-the-access-token-2

ok

Another question. If I create a custom (my) WebClient , then your GraphQLWebClient will use my WebClient or yours WebClient created by this starter,
Sorry if the question is asked awkwardly

**Comment by oliemansm** [comment, marked as answer]
You're first paragraph is correct, that is how it works. But that's the client credentials grant you're describing here. But instead of username and password it's using the client id and secret in a similar way. Make sure to read up on the various available grants and usages instead of asking that kind of details here. The https://auth0.com site has nice documentation about the various flows.

Regarding your other question. As you can see in the autoconfiguration class it'll create a WebClient instance in case it's missing. So if you're already providing a WebClient bean, then it'll use that one. That's normal Spring Boot convention. See: https://github.com/graphql-java-kickstart/graphql-spring-webclient/blob/17ffcc8a7f8dad8daf077add5eef5134a83f6fe0/graphql-webclient-spring-boot-autoconfigure/src/main/java/graphql/kickstart/spring/webclient/boot/GraphQLWebClientAutoConfiguration.java#L36.

**Comment by vadrem** [comment]
Thanks for your answers

**Reply by oliemansm** [reply]
Sure thing. Can you ask it in the Discussions section next time? Issues are really for bugs, not general questions like this one.
